pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -175,7 +175,7 @@ public void init(SegmentGeneratorConfig config, SegmentCreationDataSource dataSo
 
     // Initialize index creation
     _segmentIndexCreationInfo = new SegmentIndexCreationInfo();
-    _indexCreationInfoMap = new HashMap<>();
+    _indexCreationInfoMap = new LinkedHashMap<>(); //flaky
```

Comment:
Is there a way for you to fix the flaky test without changing the code under test?

Reply:
> The source of flakiness comes from the two HashMap and the non-determinism is not wanted by users or developers
I'm not sure how you know this fact unless you contacted developers or users or understand more about the code. Non-determinism is useful and desirable in various scenarios. Flaky tests are undesirable not because they are often related to non-determinism, but because their nondeterministic nature misleads developers about their recent changes. Unless the non-determinism in the code under test is clearly having a negative effect on developers or users, it may be preferable to simply keep the code under test nondeterministic, but ensure that the test code is deterministic. 

> 1) I can fix the code under test in an ugly way, like read the lines of the specific file "metas.properties" and sort them in a fixed order before CRC verification
I assume you mean change the test code in an ugly way here, but that may not be your only solution. You may also be able to change the way the test is checking for the lines (e.g., check for the right number of each line instead of the order of the lines).